### ISSUE TITLE
Segmentation fault when opening invalid image

### ISSUE THREAD

**Issue by khvn26** [python-pillow/Pillow#2189]
To reproduce:

1. Create a dummy file with jpg extension:

```python
temp_wrong_image_name = 'dummy.jpg'

with open(temp_wrong_image_name, 'w') as f:
    f.write("I'm a perfectly fine JPEG image file, please trust me.")
```

2. Pass it through a POST request (django-rest-framework APIClient used):

```python
# ... #

    def test_upload_wrong_image(self):
        """
        See if a non-jpeg file will get uploaded (it should not).
        """
        ## for map image
        url = reverse('walkreport-map-image', args=[self.test_report_id])

        with open(self.temp_wrong_image_name, 'rb') as temp_image_file:
            response = self.client.post(url, {'image': temp_image_file}, format='multipart')

        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)

# ... #
```

3. Try to open it with PIL.Image:

```python
def image_is_jpeg(file_obj):
    try_image = None
    try:
        try_image = Image.open(file_obj)
        try_image.verify()
    except:
        return False
    finally:
        if try_image and try_image.format != 'JPEG':
            return False
    return True

# ... #

    image = request.data[image]        # Django InMemoryUploadedFile instance
    if not image_is_jpeg(image):
        return Response(status=HTTP_400_BAD_REQUEST, data={'detail': self.ERROR_MESSAGES['NOT_VALID_JPEG'].format(image.name)})
```

This sequence results in a segmentation fault about every 10-15 tries:

```
Current thread 0x00007fcbc534d740 (most recent call first):
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/PIL/OleFileIO.py", line 400 in <module>
  File "<frozen importlib._bootstrap>", line 222 in _call_with_frames_removed
  File "<frozen importlib._bootstrap_external>", line 665 in exec_module
  File "<frozen importlib._bootstrap>", line 673 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 958 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 969 in _find_and_load
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/PIL/FpxImagePlugin.py", line 20 in <module>
  File "<frozen importlib._bootstrap>", line 222 in _call_with_frames_removed
  File "<frozen importlib._bootstrap_external>", line 665 in exec_module
  File "<frozen importlib._bootstrap>", line 673 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 958 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 969 in _find_and_load
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/PIL/Image.py", line 398 in init
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/PIL/Image.py", line 2312 in open
  File "/home/khvn/walk_reporter/walk_reporter/reports/util.py", line 7 in image_is_jpeg
  File "/home/khvn/walk_reporter/walk_reporter/reports/views.py", line 50 in map_image
  File "/usr/lib/python3.5/contextlib.py", line 30 in inner
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/rest_framework/views.py", line 474 in dispatch
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/rest_framework/viewsets.py", line 83 in view
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/views/decorators/csrf.py", line 58 in wrapped_view
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/core/handlers/base.py", line 185 in _get_response
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/core/handlers/exception.py", line 39 in inner
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/utils/deprecation.py", line 133 in __call__
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/core/handlers/exception.py", line 39 in inner
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/utils/deprecation.py", line 133 in __call__
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/core/handlers/exception.py", line 39 in inner
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/utils/deprecation.py", line 133 in __call__
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/core/handlers/exception.py", line 39 in inner
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/utils/deprecation.py", line 133 in __call__
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/core/handlers/exception.py", line 39 in inner
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/utils/deprecation.py", line 133 in __call__
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/core/handlers/exception.py", line 39 in inner
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/utils/deprecation.py", line 133 in __call__
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/core/handlers/exception.py", line 39 in inner
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/utils/deprecation.py", line 133 in __call__
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/core/handlers/exception.py", line 39 in inner
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/core/handlers/base.py", line 124 in get_response
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/rest_framework/test.py", line 260 in get_response
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/test/client.py", line 142 in __call__
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/test/client.py", line 476 in request
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/rest_framework/test.py", line 240 in request
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/rest_framework/test.py", line 288 in request
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/django/test/client.py", line 409 in generic
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/rest_framework/test.py", line 221 in post
  File "/home/khvn/.virtualenvs/reporter/lib/python3.5/site-packages/rest_framework/test.py", line 299 in post
```

System info:
```
$ cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=14.04
DISTRIB_CODENAME=trusty
DISTRIB_DESCRIPTION="Ubuntu 14.04.5 LTS"
$ python -V
Python 3.5.2
$ python -m django --version 
1.10.2
```

Couldn't reproduce this opening the file directly with ```open()```, though.

**Comment by wiredfool** [comment]
That's a really strange callstack.  The segfault is when OleFileIO is getting imported, and it's happening from a frozen item, which I've never seen outside of py2exe on windows. And you're on Ubuntu. 

It doesn't look related to the image (fake or otherwise), and doesn't really look related to Pillow, as it's failing on import when the vars() function is called in the module. The fact that you can't reproduce this with just pillow supports that conclusion as well. 

I'd almost believe that a .pyc file is corrupted somewhere, but I'm really not sure.

**Comment by wiredfool** [comment]
Feel free to reopen if you can isolate this to pillow.

**Comment by evanlesmez** [comment]
Also am getting this every 10ish tries. Python execution breaks and all I get is ```[1]    184358 segmentation fault (core dumped)```

**Comment by wiredfool** [comment]
Can you open a new issue with enough details to reproduce the issue?

**Comment by evanlesmez** [comment]
Figured it out is was actually an issue with making repeated calls to `ImageFont.truetype(FONT_FILE, 15)` and `ImageFont.getsize` inside a 'while` loop.

python version 3.5.6
Pillow version 7.0.0

Was able to reproduce 
```python
from PIL import ImageFont

text = ":D"
font = ImageFont.truetype(FONT_FILE, 15)
print(font.getsize(text))
time.sleep(1)
```
OUTPUT:
```sh
(16, 14)
corrupted size vs. prev_size in fastbins
[1]    13629 abort (core dumped)  python -m cam.__init_
```
I used an `arial.ttf` font file but I could not drag it in here unfortunately. I have not tested with other fonts. Also logically it does not make a lot of sense to instantiate a reused font Class over and over again in a while loop anyway so this is pretty minor.
